### PR TITLE
Calculate span duration using the monotonic clock

### PR DESCRIPTION
**Pull request by ramonza**
Fixes: #381

### CODE REVIEW

File: zpages/tracez.go [hunk 1]
```diff
@@ -251,10 +251,10 @@ func traceRows(s *trace.SpanData) []traceRow {
 	}
 
 	firstRow := traceRow{Fields: [3]string{wholeTime(start), "", ""}, SpanContext: s.SpanContext, ParentSpanID: s.ParentSpanID}
-	if s.EndTime.IsZero() {
+	if s.Duration == time.Duration(0) {
```
Comment (rakyll) [inline zpages/tracez.go]: s.Duration == 0

File: trace/export.go [hunk 2]
```diff
@@ -56,16 +56,22 @@ func UnregisterExporter(e Exporter) {
...
+	Annotations     []Annotation
+	MessageEvents   []MessageEvent
+	Status          Status
 	Links           []Link
 	HasRemoteParent bool
 }
+
+// EndTime calculates the Duration offset from StartTime. This is robust to wall
+// clock changes between the start and end of the span.
+func (s *SpanData) EndTime() time.Time {
```
Comment (rakyll) [inline trace/export.go]: I don't think we need to provide this. We can simply calculate it from start+dur.

**Comment by ramonza** [comment]
Updated to restore EndTime and just adjust it.

**Comment by ramonza** [comment]
@rakyll PTAL

**Pull request merged**